### PR TITLE
Return an empty list in AWS's _on_list_ready on a timeout response

### PR DESCRIPTION
**Pull request by physicalattraction**
This would circumvent the problems described here: https://github.com/celery/kombu/issues/931

I am unaware of the testing procedures in this library. Can someone check if this is a good approach?

### CODE REVIEW

File: kombu/asynchronous/aws/connection.py [hunk 1]
```diff
@@ -170,6 +170,11 @@ def _mexe(self, request, sender=None, callback=None):
 class AsyncAWSQueryConnection(AsyncConnection):
     """Async AWS Query Connection."""
 
+    STATUS_CODE_OK = 200
+    STATUS_CODE_REQUEST_TIMEOUT = 408
+    STATUS_CODE_NETWORK_CONNECT_TIMEOUT_ERROR = 599
+    STATUS_CODES_TIMEOUT = (STATUS_CODE_REQUEST_TIMEOUT, STATUS_CODE_NETWORK_CONNECT_TIMEOUT_ERROR)
```
Comment (physicalattraction) [inline kombu/asynchronous/aws/connection.py]: What's the benefit of a tuple instead of a list?

**Comment by physicalattraction** [comment]
Can you explain how you run the tests? When I run them either in PyCharm or in the terminal, there no tests to execute.

```
$ python -m unittest t.unit.asynchronous.aws.sqs.test_connection

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

File: kombu/asynchronous/aws/connection.py (hunk 1 above)
Comment (thedrow) [reply kombu/asynchronous/aws/connection.py]: Using a tuple results in a smaller memory footprint. They are also faster to allocate on CPython as far as I recall.
Granted, it's only two items but there's no need to use more memory when you can avoid it.

**Comment by thedrow** [comment]
We use pytest instead of unittest.
Check https://github.com/celery/kombu/blob/master/tox.ini#L28 for the exact command run in CI.
You could also use tox yourself.
Also, please rebase. The tests in master should pass now.

**Comment by physicalattraction** [comment]
> Please add test coverage for the new code branch.

@thedrow: I have no clue how to test this. I don't see any existing test case in which this method is called. Can you give me a clue?

> could you please check the flake8 error?

What is flake8?

**Comment by thedrow** [comment]
You can run the tests using [tox](https://tox.readthedocs.io/en/latest/).

**Comment by physicalattraction** [comment]
Running the tests is not a problem, I did this with the command `pytest -rxs -xv`. The problem I am facing is that I am unfamiliar with the test structure in this repository. I added a fix in one particular method, but I don't see any existing unit tests for this method.

I have ran flake8 and in the latest commit made sure that there are no additional warnings compared to the master branch.

**Comment by thedrow** [comment]
You are correct. There are no tests for `get_list()`.
All you need to see is to mock `transform()` and ensure it uses the correct values in your case.
See https://github.com/celery/kombu/blob/c00be52bce3aa1a32a9344dc75e26347b99f96d4/kombu/asynchronous/aws/connection.py#L211

**Comment by thedrow** [comment]
I'm going to merge this and open an issue about writing tests for this later.
I think this is a proper fix for the problem.

**Pull request merged**